PyRDF/DataFrame.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import print_function
+
+import logging
+
+import ROOT
+
+from PyRDF import Node, Proxy
+from PyRDF.backend import Spark
+
+logger = logging.getLogger(__name__)
+
+
+class SparkDataFrame(object):
```

Comment:
The problem with this is that we are introducing backend-dependent code in a part of the PyRDF code that should be agnostic to that.

I wonder if we could do this differently. In `PyRDF/backends` lives everything that has to do with backends. We could import whatever is in `PyRDF/backends` when importing PyRDF, and those imports would dynamically inject in PyRDF the calls that allow to create RDFs for their backends. We do a similar thing for pythonizations in ROOT:

https://github.com/root-project/root/blob/master/bindings/pyroot/pythonizations/python/ROOT/__init__.py#L67-L69

The `ROOT` module does not know which pythonizations exist, but it knows they are in a subfolder `pythonization`. By importing the modules in `pythonization`, the available pythonizors for each class are registered. If a new pythonizor is added to that folder, it is automatically taken by ROOT.

Similarly, we could follow a convention so that the backends do an injection in their `__init__.py` file of their factory function. If a new backend is added, we would not need to change backend-independent code.

Reply:
Interesting, looking into this :+1: